source/MaterialXGenShader/SgNode.cpp

```diff
@@ -482,25 +482,54 @@ void SgNodeGraph::addUpstreamDependencies(const Element& root, ConstMaterialPtr
     }
 }
 
-void SgNodeGraph::addDefaultGeomNode(SgInput* input, const string& geomNode, ShaderGenerator& shadergen)
+void SgNodeGraph::addDefaultGeomNode(SgInput* input, const GeomProp& geomprop, ShaderGenerator& shadergen)
 {
-    const string geomNodeName = "default_" + geomNode;
+    const string geomNodeName = "default_" + geomprop.getName();
     SgNode* node = getNode(geomNodeName);
 
     if (!node)
     {
-        string geomNodeDefName = "ND_" + geomNode + "_" + input->type->getName();
+        string geomNodeDefName = "ND_" + geomprop.getName() + "_" + input->type->getName();
         NodeDefPtr geomNodeDef = _document->getNodeDef(geomNodeDefName);
         if (!geomNodeDef)
         {
             throw ExceptionShaderGenError("Could not find a nodedef named '" + geomNodeDefName +
-                "' for defaultgeomprop on input '" + input->node->getName() + "." + input->name + "'");
+                "' for geomprop on input '" + input->node->getName() + "." + input->name + "'");
         }
 
         SgNodePtr geomNodePtr = SgNode::create(geomNodeName, *geomNodeDef, shadergen);
         _nodeMap[geomNodeName] = geomNodePtr;
         _nodeOrder.push_back(geomNodePtr.get());
 
+        // Set node inputs if given.
+        const string& space = geomprop.getSpace();
+        if (!space.empty())
+        {
+            SgInput* spaceInput = geomNodePtr->getInput("space");
+            if (spaceInput)
+            {
+                spaceInput->value = Value::createValue<string>(space);
+            }
+        }
+        const string& index = geomprop.getIndex();
+        if (!index.empty())
+        {
+            SgInput* indexInput = geomNodePtr->getInput("index");
+            if (indexInput)
+            {
+                indexInput->value = Value::createValue<string>(index);
+            }
+        }
+        const string& attrname = geomprop.getAttrName();
+        if (!attrname.empty())
+        {
+            SgInput* attrnameInput = geomNodePtr->getInput("attrname");
+            if (attrnameInput)
+            {
+                attrnameInput->value = Value::createValue<string>(attrname);
+            }
+        }
```

Comment:
Should we throw an exception here if there is a geomprop with no name?

Reply:
If the geomprop doesn't have a name this would be caught by the search for a nodedef above, which is throwing an exception already. But not sure if you meant something else?

Reply:
Maybe I'm confused about the usage of "attrname" vs the geomprop name. I assume attrname the name of actual attribute you want. 
So for example this is possible:
```
<geomprop name="normal1" attrname="normal"/>
```
Thus the of the geomprop can be anything (doesn't need to be an attribute name) and attrname is therefore required.

Reply:
Cleared up with Niklas that this is for geomattrvalue.